source/codegen/templates/service_helpers.mako

```diff
@@ -659,20 +657,29 @@ ${copy_to_response_with_transform(source_buffer=raw_response_field, parameter_na
 %   elif service_helpers.is_output_array_that_needs_coercion(parameter):
 ${initialize_response_buffer(parameter_name=parameter_name, parameter=parameter)}\
 ${copy_to_response_with_transform(source_buffer=parameter_name, parameter_name=parameter_name, transform_x="x")}\
+%   elif common_helpers.is_struct(parameter) or parameter['type'] == 'ViBoolean[]':
+        convert_to_grpc(${parameter_name}, response->mutable_${parameter_name}());
+%   elif common_helpers.is_string_arg(parameter):
+        response->set_${parameter_name}(${parameter_name});
+%   elif parameter['type'] == 'ViSession':
+        auto session_id = session_repository_->resolve_session_id(${parameter_name});
+        response->mutable_${parameter_name}()->set_id(session_id);
+%   elif parameter['type'] == 'CVIAbsoluteTime':
+        convert_to_grpc(${parameter_name}, response->mutable_${parameter_name}());
 %   elif common_helpers.is_array(parameter['type']):
-%     if common_helpers.is_string_arg(parameter):
+### pass: other array types don't need to copy.
+%   else:
         response->set_${parameter_name}(${parameter_name});
-%     elif common_helpers.is_struct(parameter) or parameter['type'] == 'ViBoolean[]':
-        Copy(${parameter_name}, response->mutable_${parameter_name}());
-%     endif
+%   endif
+### Handle ivi-dance-with-a-twist resizing.
```

Comment:
The `elif`s get broken by this change if I leave this ivi-twist-resize in the original switch statement. (actually they were broken before, but for a corner case: bytes enums). So I moved it to a separate if block after the initial copy.